src/components/Content.jsx

```diff
@@ -0,0 +1,45 @@
+import { useState, useEffect } from "react";
+
+const TOTAL_OF_PAGES = 4;
+const POSTS_PER_PAGE = 20;
+const NUMBER_OF_POSTS = 95;
+
+const Content = () => {
+  const [posts, setPosts] = useState([]);
+  const [error, setError] = useState(null);
+
+  const loadData = () => {
+    const startIndex = posts.length;
+    const endIndex = startIndex + (posts.length < 80 ? POSTS_PER_PAGE : 15);
+    fetch(`https://jsonplaceholder.typicode.com/posts?_start=${startIndex}&_end=${endIndex}`)
+      .then((response) => {
+        response.json().then((data) => {
+          const updatedPosts = posts.concat(data);
+          setPosts(updatedPosts);
+        })
+      })
+      .catch((e) => {
+        setError(e)
+      });
+  }
+
+  useEffect(() => {
+    if(posts.length < NUMBER_OF_POSTS){
+      loadData();
+    }
+  });
```

Comment:
What do you think about not depending on React to manage when a new request should be triggered? The code we have right now depends on React to re-render the component to trigger the next request, but data fetching is not React's responsibility. Can we move the load of the posts to a function that is called only once in the component and only resolves the promise once all of them are loaded?